snapcraft/internal/elf.py

```diff
@@ -91,27 +94,60 @@ def _get_system_libs():
     return _libraries
 
 
-def get_dependencies(elf):
-    """Return a list of libraries that are needed to satisfy elf's runtime.
+def get_dependencies(elf: str) -> Set[str]:
+    """Return a set of libraries that are needed to satisfy elf's runtime.
 
     This may include libraries contained within the project.
     """
     logger.debug('Getting dependencies for {!r}'.format(str(elf)))
-    ldd_out = ''
+    ldd_out: List[str] = []
     try:
         ldd_out = common.run_output(['ldd', elf]).split('\n')
     except subprocess.CalledProcessError:
         logger.warning(
             'Unable to determine library dependencies for {!r}'.format(elf))
-        return []
-    ldd_out = [l.split() for l in ldd_out]
-    ldd_out = [l[2] for l in ldd_out if len(l) > 2 and os.path.exists(l[2])]
+        return set()
+    ldd_out_split = [l.split() for l in ldd_out]
+    ldd_out_list = [l[2] for l in ldd_out_split
+                    if len(l) > 2 and os.path.exists(l[2])]
 
     # Now lets filter out what would be on the system
     system_libs = _get_system_libs()
-    libs = [l for l in ldd_out if not os.path.basename(l) in system_libs]
+    libs = {l for l in ldd_out_list if not os.path.basename(l) in system_libs}
     # Classic confinement won't do what you want with library crawling
     # and has a nasty side effect described in LP: #1670100
-    libs = [l for l in libs if not l.startswith('/snap/')]
+    libs = {l for l in libs if not l.startswith('/snap/')}
 
     return libs
+
+
+def get_elf_files(root: str, file_list: Sequence[str]) -> FrozenSet[str]:
+    """Return a frozenset of elf files from file_list prepended with root."""
+    ms = magic.open(magic.NONE)
+    if ms.load() != 0:
+        raise RuntimeError('Cannot load magic header detection')
+
+    elf_files = set()
+
+    fs_encoding = sys.getfilesystemencoding()
+
+    for part_file in file_list:
+        # Filter out object (*.o) files-- we only care about binaries.
+        if part_file.endswith('.o'):
+            continue
+
+        # No need to crawl links-- the original should be here, too.
+        path: str = os.path.join(root, part_file)
```

Comment:
According to [this](https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html), variable annotations (PEP 526) were added in Python 3.6. Given that Xenial is only 3.5.2, that will probably be an issue.

Reply:
Yeah, Travis is barfing.